pkg/channel/distributed/controller/kafkachannel/dispatcher.go

```diff
@@ -144,20 +148,35 @@ func (r *Reconciler) reconcileDispatcherService(ctx context.Context, logger *zap
 		}
 	} else {
 
+		// Determine whether the existing service is different in a way that demands a patch
+		// such as missing required labels or spec differences
+		patch, needsUpdate := util.CheckServiceChanged(logger, existingService, newService)
+
```

Comment:
Can this be moved down a bit (after deletion timestamp check just before `if needsUpdate`?  Just in the spirit of declaring things close to where used.

Reply:
But of course